x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java

```diff
@@ -182,25 +188,27 @@ protected void doExecute(
             );
         }));
 
-        // When the request resource is a deployment find the
-        // model used in that deployment for the model stats
-        String idExpression = addModelsUsedInMatchingDeployments(request.getResourceId(), assignmentMetadata);
-        logger.debug("Expanded models/deployment Ids request [{}]", idExpression);
-
-        // the request id may contain deployment ids
-        // It is not an error if these don't match a model id but
-        // they need to be included in case the deployment id is also
-        // a model id. Hence, the `matchedDeploymentIds` parameter
-        trainedModelProvider.expandIds(
-            idExpression,
-            request.isAllowNoResources(),
-            request.getPageParams(),
-            Collections.emptySet(),
-            modelAliasMetadata,
-            parentTaskId,
-            matchedDeploymentIds,
-            idsListener
-        );
+        executor.execute(() -> {
```

Comment:
I'd rather we used an `ActionRunnable` here too for better exception-safety and also in case `executor` ever became something that might reject tasks on shutdown/overload:

```suggestion
        executor.execute(ActionRunnable.wrap(idsListener, l -> {
```

(also avoiding capture of `idsListener` below)